samples/wirelog/wirelog_test.go

```diff
@@ -0,0 +1,201 @@
+// Copyright 2025 Google Inc. All Rights Reserved.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package wirelog
+
+import (
+	"bytes"
+	"context"
+	"encoding/json"
+	"os"
+	"path"
+	"syscall"
+	"testing"
+	"time"
+
+	"github.com/jacobsa/fuse"
+	"github.com/jacobsa/fuse/fuseops"
+	. "github.com/jacobsa/ogletest"
+)
+
+func TestWireLog(t *testing.T) { RunTests(t) }
+
+type WireLogTest struct {
+	ctx context.Context
+	dir string
+	mfs *fuse.MountedFileSystem
+	buf bytes.Buffer
+}
+
+func init() { RegisterTestSuite(&WireLogTest{}) }
+
+func (t *WireLogTest) SetUp(ti *TestInfo) {
+	t.ctx = ti.Ctx
+	var err error
+	t.dir, err = os.MkdirTemp("", "wirelog_test")
+	AssertEq(nil, err)
+
+	// Mount the file system.
+	t.mfs, err = fuse.Mount(t.dir, NewTestFS(), &fuse.MountConfig{
+		WireLogger: &t.buf,
+		OpContext:  t.ctx,
+	})
+	AssertEq(nil, err)
+}
+
+func (t *WireLogTest) TearDown() {
+	// Ensure unmounted.
+	if t.mfs != nil {
+		fuse.Unmount(t.dir)
+		t.mfs.Join(t.ctx)
+	}
+	os.RemoveAll(t.dir)
+}
+
+// Helper to load Args into a struct
+func loadArgs(entry fuse.WireLogRecord, dst any) {
+	b, err := json.Marshal(entry.Args)
+	AssertEq(nil, err)
+	err = json.Unmarshal(b, dst)
+	AssertEq(nil, err)
+}
+
+func (t *WireLogTest) RunWorkloadAndCheckLogs() {
+	// 1. Stat the file.
+	filePath := path.Join(t.dir, "foo")
+	fi, err := os.Stat(filePath)
+	AssertEq(nil, err)
+	ExpectEq(3, fi.Size())
+
+	// 2. Read the file.
+	content, err := os.ReadFile(filePath)
+	AssertEq(nil, err)
+	ExpectEq("bar", string(content))
+
+	// Unmount to ensure everything is flushed/closed.
+	err = fuse.Unmount(t.dir)
+	AssertEq(nil, err)
+
+	// Wait for the connection to close.
+	err = t.mfs.Join(t.ctx)
+	AssertEq(nil, err)
+
+	// Mark as joined so TearDown doesn't try again.
+	t.mfs = nil
+
+	// Parse the logs.
+	ops := make(map[string][]fuse.WireLogRecord)
+	decoder := json.NewDecoder(&t.buf)
+
+	for decoder.More() {
+		var entry fuse.WireLogRecord
+		err := decoder.Decode(&entry)
+		AssertEq(nil, err)
+		ExpectTrue(time.Now().After(entry.StartTime))
+		ExpectGt(entry.Duration, 0)
+		ops[entry.Operation] = append(ops[entry.Operation], entry)
+	}
+
+	// 1. initOp
```

Comment:
I know that this project unfortunately uses an assertion library for testing, but I do wonder if using `cmp.Diff` wouldn’t be easier to read here… Can you give that a try please?

Reply:
I added this as a separate commit. Please have a look. I'm not super convinced this is better because 1/ the structs have many fields that we don't want to compare, so I had to add a "ignoreZeros" filter, and 2/ there are some greater/less than/not nil/zero checks that you still have to do as separate checks, 3/ we can't match the overall wirelog record in one go because Args is a map[string]any that needs to be loaded separately.